### PR TITLE
Get post response

### PR DESCRIPTION
**Pull request by gastrodon**
The purpose of this branch is to add functionality of getting references to posts or comments POSTed to Reddit when they are created. For example this is useful for bots that post content that requires a description comment or a flare to be added at post time. Because Reddit does not return full post objects when submitted, the Submission type is only meant to contain a name, id, and url to access those created resources normally.

This is done by adding account methods `GetReply`, `GetPostSelf`, and `GetPostLink` that use a new sow method `get_sow`, which itself uses a new parser method `parse_submitted`. The existing parsing method was not used because of differences in how Reddit formats json requests made with suffix `.json` and the url_param `api_type=json` that these new methods use.

Tests have been written for added methods, and mock types implement mock instances of these methods where they should.

### CODE REVIEW

**Comment by turnage** [comment]
Would you mind revising the git history? Many of these commits edit or reverse each other.

You can do this with `git rebase HEAD~15 -i` to re-order, squash, etc.

**Comment by gastrodon** [comment]
Would a squash and merge not be preferable for a feature branch? I'm not quite sure how exactly to go about revising this history, that's how I usually handle branches being merged into some master.

**Comment by gastrodon** [comment]
Hello @turnage, the commit history has been revised. My last comment can be disregarded. Because of the messy process I took in this branch, I opted to re-commit most of the files and overwrite this branch

File: reddit/parse.go [hunk 1]
```diff
@@ -67,6 +68,37 @@ func (p *parserImpl) parse(
...
+	if len(wrapped["errors"].([]interface{})) != 0 {
+		return Submission{}, fmt.Errorf("API errors were returned: %v", wrapped["errors"])
+	}
+
+	data := wrapped["data"].(map[string]interface{})
+
+	// Comment submissions are further wrapped in a things block
+	// because of ... something? There only appears to be a single thing
+	// This transformes var data to be the data of the single thing
+	// This also mirrors the key permalink -> url because they are functionally the same
```
Comment (turnage) [inline reddit/parse.go]: Does permalink include the scheme and domain, or is just the path? I would expect something called "url" to contain the scheme and domain.

File: reddit/reaper.go [hunk 2]
```diff
@@ -95,6 +98,25 @@ func (r *reaperImpl) sow(path string, values map[string]string) error {
...
+			Method: "POST",
+			Header: formEncoding,
+			Host:   r.hostname,
+			URL:    r.url(path, values),
+		},
+	)
+
+	if err != nil {
+		return Submission{}, err
+	}
```
Comment (turnage) [inline reddit/reaper.go]: Would you also check here for non-200 return code? `err` may be nil even if Reddit decided this was an error. If you call `parse_submitted` with the response body, a 400 error for example would look like a json parsing error.

File: reddit/parse.go (hunk 1 above)
Comment (gastrodon) [reply reddit/parse.go]: Yes, I missed that. That block has been updated to prefix the permalink, though I wonder if there's a better way that I should be going about that

File: reddit/reaper.go (hunk 2 above)
Comment (gastrodon) [reply reddit/reaper.go]: Does `baseClient.Do` not already check for non-200 status codes [here](https://github.com/gastrodon/graw/blob/get-post-response/reddit/client.go#L41)?

Comment (turnage) [reply reddit/reaper.go]: My mistake; I forgot that was `baseClient` and not `http.Client`.

File: reddit/parse.go (hunk 1 above)
Comment (turnage) [reply reddit/parse.go]: I think your solution is fine.

If you squash this commit into `Create parse_submitted` I will merge.

Comment (gastrodon) [reply reddit/parse.go]: History was updated

**Comment by turnage** [comment]
Thank you for contributing! Please help yourself to contributors.md if you would like.

**Pull request merged**